src/System.Net.WebSockets.Client/tests/SendReceiveTest.cs

```diff
@@ -86,55 +85,44 @@ public class SendReceiveTest : ClientWebSocketTestBase
             {
-                var cts = new CancellationTokenSource(TimeOutMilliseconds);
+                var tasks = new Task[2];
 
-                Task[] tasks = new Task[10];
+                // Need large buffer, or Send completes too quickly and no exception is thrown.
+                var sendBuffer = new byte[128 * 1024];
+                var segment = new ArraySegment<byte>(sendBuffer);
 
-                try
+                for (int i = 0; i < tasks.Length; i++)
                 {
-                    for (int i = 0; i < tasks.Length; i++)
-                    {
-                        tasks[i] = cws.SendAsync(
-                            WebSocketData.GetBufferFromText("hello"),
-                            WebSocketMessageType.Text,
-                            true,
-                            cts.Token);
-                    }
+                    tasks[i] = cws.SendAsync(
+                        segment,
+                        WebSocketMessageType.Text,
+                        true,
+                        CancellationToken.None);
+                }
 
-                    Task.WaitAll(tasks);
+                await Assert.ThrowsAsync<InvalidOperationException>(() => Task.WhenAll(tasks));
+                Assert.Equal(WebSocketState.Aborted, cws.State);
 
-                    Assert.Equal(WebSocketState.Open, cws.State);
-                }
-                catch (AggregateException ag)
+                Assert.All(tasks, task =>
                 {
-                    foreach (var ex in ag.InnerExceptions)
-                    {
-                        if (ex is InvalidOperationException)
-                        {
-                            Assert.Equal(
-                                ResourceHelper.GetExceptionMessage(
-                                    "net_Websockets_AlreadyOneOutstandingOperation",
-                                    "SendAsync"),
-                                ex.Message);
-
-                            Assert.Equal(WebSocketState.Aborted, cws.State);
-                        }
-                        else if (ex is WebSocketException)
-                        {
-                            // Multiple cases.
-                            Assert.Equal(WebSocketState.Aborted, cws.State);
-
-                            WebSocketError errCode = (ex as WebSocketException).WebSocketErrorCode;
-                            Assert.True(
-                                (errCode == WebSocketError.InvalidState) || (errCode == WebSocketError.Success),
-                                "WebSocketErrorCode");
-                        }
-                        else
-                        {
-                            Assert.IsAssignableFrom<OperationCanceledException>(ex);
-                        }
-                    }
-                }
+                    Assert.Equal(TaskStatus.Faulted, task.Status);
```

Comment:
Hmmm, well, you're right that they can complete (which looked to have happened on Linux), if they manage to send all the data in the buffer before the second `Send` is called.  This was why I increased the size of the data sent, although I was aware it might not be completely future-proof.

The way to fix it, though, would be to tell the echo server to not accept the whole thing, or delay the send somehow.  Need to look into it more, but my first guess would be having it read/parse by token, instead of the entire message, might do it.